src/Compilers/Core/Portable/Operations/BasicBlock.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    /// <summary>
+    /// PROTOTYPE(dataflow): Add documentation
+    /// </summary>
+    public enum BasicBlockKind
+    {
+        Entry,
+        Exit,
+        Block
+    }
+
+    /// <summary>
+    /// PROTOTYPE(dataflow): Add documentation
+    /// </summary>
+    public sealed class BasicBlock
+    {
+        private ImmutableArray<IOperation>.Builder _statements;
+        private ImmutableHashSet<BasicBlock>.Builder _predecessors;
+
+        public BasicBlock(BasicBlockKind kind)
+        {
+            Kind = kind;
+            _statements = ImmutableArray.CreateBuilder<IOperation>();
+            _predecessors = ImmutableHashSet.CreateBuilder<BasicBlock>();
+        }
+
+        public BasicBlockKind Kind { get; private set; }
+        public ImmutableArray<IOperation> Statements => _statements.ToImmutable();
```

Comment:
Could the block be marked as frozen at this point? If so, `Statements` could be cached.